db/txn_db.go

```diff
@@ -174,14 +174,154 @@ func (db *implicitTxnDB) GetAllIndexes(
 	}
 	defer txn.Discard(ctx)
 
-	return db.getAllIndexes(ctx, txn)
+	return db.getAllIndexDescriptions(ctx, txn)
 }
 
 // GetAllIndexes gets all the indexes in the database.
 func (db *explicitTxnDB) GetAllIndexes(
 	ctx context.Context,
 ) (map[client.CollectionName][]client.IndexDescription, error) {
-	return db.getAllIndexes(ctx, db.txn)
+	return db.getAllIndexDescriptions(ctx, db.txn)
+}
+
+// CreateDocIndex creates a new index for the given document.
+func (db *implicitTxnDB) CreateDocIndex(
+	ctx context.Context,
+	col client.Collection,
+	doc *client.Document,
+) error {
+	txn, err := db.NewTxn(ctx, true)
+	if err != nil {
+		return err
+	}
+	defer txn.Discard(ctx)
+
+	indexes, err := db.getCollectionIndexes(ctx, txn, col)
+	if err != nil {
+		return err
+	}
+
+	for _, index := range indexes {
```

Comment:
todo: Please don't include any non transaction related logic in this file - instead move this block into a `db.createDocIndex` func (along with the others).

This allows us to keep the transaction boiler-plate stuff separate from the more complex business logic.

Reply:
done